command/commands/token.py

```diff
@@ -0,0 +1,77 @@
+"""Command to obtain signed authentication token."""
+import jwt
+import logging
+
+from datetime import datetime
+from model.permissions import Permissions
+
+
+class TokenCommand:
+    """Token command model class."""
+
+    command_name = "token"
+    desc = "Generate a signed token for use with the HTTP API"
+    permission_error = "You do not have the sufficient " \
+                       "permission level for this command!"
+    lookup_error = "Requesting user not found!"
+
+    def __init__(self, db_facade, config):
+        """
+        Initialize TokenCommand.
+
+        :param db_facade: Database connection
+        """
+        logging.info("Initializing TokenCommand instance")
+        self.facade = db_facade
+        self.expiry = config.get_expiry()
+        self.signing_key = config.get_signing_key()
+
+    def get_name(self):
+        """Return the command's name."""
+        return self.command_name
+
+    def get_desc(self):
+        """Return a description of the command."""
+        return self.desc
```

Comment:
this is more of an aside, but TIL lots of getters is a thing in python, I thought the meme was that java's getters are half of its verbosity?

Reply:
I'm not sure that a lot of getters is actually a thing in Python - thoughts, @cheukyin699?

Reply:
I think he's referring to the `@property` decorator, where you could make a function with that decorator and have slightly easier access to the variables within. Also see `@var.setter` and `@var.getter`.

A slight caveat is the classes that implement this must be children of the mighty `object` class. But nothing a bit of refactoring can't do.

Reply:
Sounds like another ticket. 😛